server/data/data-library/src/main/java/org/apache/james/rrt/lib/AbstractRecipientRewriteTable.java

```diff
@@ -263,9 +263,6 @@ public void addMapping(String user, Domain domain, String mapping) throws Recipi
         } else if (mappingType.equals(Type.Regex)) {
             addRegexMapping(user, domain, mappingSuffix);
         } else if (mappingType.equals(Type.Domain)) {
-            if (user != null) {
```

Comment:
I don't like this decision : how about adding a log because we found something that is not consistent ?

Reply:
Note that WILDCARD user also make sens here IMO